util/site/build-docs.sh

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+
+set -e
+
+command="build"
+
+case "$1" in
+  "build"|"build-local"|"serve"|"serve-proxy")
+    command="$1"
+    public_url="${2}"
+    public_port="${3}"
+    ;;
+  "help"|*)
+    echo "USAGE: $0 [command]"
+    echo ""
+    echo "commands:"
+    echo "  help         prints this message."
+    echo "  build        build the site and docs for prod"
+    echo "  build-local  build the site and docs for a localhost server"
+    echo "  serve        build and serve the site locally"
+    echo "  serve-proxy  build and serve the site, with a public url"
+    exit 0
+    ;;
+esac
+
+################
+# DEPENDENCIES #
+################
+
+checkDeps () {
+    # Check for mdbook dep
+    if ! command -v mdbook >/dev/null; then
+        echo "E: mdbook not found, please install from your package manager or with:" >&2
+        echo "E:   $ cargo install mdbook" >&2
+        exit 1
+    fi
+
+    # Check for hugo dep
+    if ! command -v hugo >/dev/null; then
+        echo "E: hugo not found, please install from your package manager" >&2
+        exit 1
+    fi
+}
+checkDeps
+
+#################
+# CONFIGURATION #
+#################
+declare base_url
+declare docs_url
+declare serve_port
+declare public_port
+
+# Get the project directory from the location of this script
+this_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+proj_root=$(realpath "${this_dir}/../..")
+
+# Create the output directory
+build_dir="$proj_root/build-site"
+mkdir -p "$build_dir"
+
+getURLs () {
+    # Default urls here are for production
+    local base_stem="https://opentitan.org"
+
+    # Use localhost for URL's when building/serving locally
+    if [ "$command" = "build-local" ] || \
+       [ "$command" = "serve" ]; then
+        base_stem="http://localhost"
+        serve_port=":9000"
+        public_port=":9000"
+    fi
+    # If the site is behind a proxy, set the URL/public_port appropriately
+    if [ "$command" = "serve-proxy" ] ; then
+        base_stem="http://${public_url}"
+        serve_port=":8000"
+        public_port=":${public_port}"
+    fi
+
+    base_url="${base_stem}${public_port}"
+    docs_url="${base_stem}${public_port}/book"
+}
+getURLs
+
+# Export some environment variables that tools will pick up
+export HUGO_PARAMS_DOCSURL="${docs_url}" # hugo
+export URL_ROOT="${docs_url}" # earlgrey_diagram
+
+# Build up doxygen command
+doxygen_env="env"
+doxygen_env+=" SRCTREE_TOP=${proj_root}"
+doxygen_env+=" DOXYGEN_OUT=${build_dir}/gen"
```

Comment:
Mixture of `export` and `env` - I think the `export`s are easier to read personally.

Reply:
Yeah, I think `export`s are easier to read. I like `env` because it keeps environment variables scoped to the command they affect. So, I'm ok with both co-existing, but don't have particularly strong opinions either way.